src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -3097,7 +3096,7 @@ tx_gateway_frontend::get_all_transactions_for_one_tx_partition(
                 return_all_txs_res{tx_errc::unknown_server_error});
           }
 
-          auto gate_lock = gate_guard(self._gate);
+          auto gate_lock = self._gate.hold();
```

Comment:
Not sure what this holder does. I don't see it used below.

Reply:
I had the same question: https://github.com/redpanda-data/redpanda/pull/13545

Reply:
I think that this method used to be a coroutine before a refactoring

https://github.com/redpanda-data/redpanda/blame/14482249f1f6c3e284534459591bec452c8d7cc9/src/v/cluster/tx_gateway_frontend.cc#L1216

and this probably just got left in here. kinda seems like the gate _should_ be captured.